packages/opentelemetry-cloud-monitoring-exporter/src/transform.ts

```diff
@@ -147,37 +142,58 @@ function transformPoint(
   // TODO: Add endTime and startTime support, once available in OpenTelemetry
   // Related issues: https://github.com/open-telemetry/opentelemetry-js/pull/893
   // and https://github.com/open-telemetry/opentelemetry-js/issues/488
-  if (metricDescriptor.metricKind === OTMetricKind.COUNTER) {
-    return {
-      value: transformValue(metricDescriptor.valueType, point.value),
-      interval: {
-        startTime,
-        endTime: new Date().toISOString(),
-      },
-    };
+  switch (metricDescriptor.metricKind) {
+    case OTMetricKind.COUNTER:
+    case OTMetricKind.SUM_OBSERVER:
+      return {
+        value: transformValue(metricDescriptor.valueType, point.value),
+        interval: {
+          startTime,
+          endTime: new Date().toISOString(),
+        },
+      };
+    default:
+      return {
+        value: transformValue(metricDescriptor.valueType, point.value),
+        interval: {
+          endTime: new Date().toISOString(),
+        },
+      };
   }
-  return {
-    value: transformValue(metricDescriptor.valueType, point.value),
-    interval: {
-      endTime: new Date().toISOString(),
-    },
-  };
 }
 
 /** Transforms a OpenTelemetry Point's value to a StackDriver Point value. */
 function transformValue(
   valueType: OTValueType,
-  value: number | OTDistribution
+  value: number | OTDistribution | OTHistogram
 ) {
+  if (isDistributionValue(value)) {
+    // TODO: Add support for Distribution
+    throw new Error('Distributions are not yet supported');
+  }
+  if (isHistogramValue(value)) {
+    // TODO: Add support for Histogram
+    throw new Error('Histograms are not yet supported');
+  }
```

Comment:
This is inconsistent with the existing code, which uses the `valueType` rather than the value to distinguish the behavior. I think there are pros and cons to each; using the `valueType` isn't as type safe as using the `value`, but it is simpler. I would recommend using `valueType` for now and perhaps revisiting when distributions/histograms are implemented. Does that make sense to you? Having explicit cases for each with TODOs and specific errors still looks good to me.

Reply:
I agree that it would be a better solution to just switch on `valueType` if `valueType` could be either int, double, histogram, or distribution. From what I understand however, `valueType` can only ever be int or double (see [link to code](https://github.com/open-telemetry/opentelemetry-js/blob/904245f7678537d9f3ae1ce31bd7a8b9469cc7db/packages/opentelemetry-api/src/metrics/Metric.ts#L64-L67)). I believe that switching solely on `valueType` would cause unexpected behaviour when `value` is a `Distribution` or `Histogram`. For example when `valueType` is `INT` and `value` is a `Distribution`, the output of the function would have the following structure: 
```
{
  int64Value: {
    min: <int>,
    max: <int>,
    count: <int>,
    sum: <int>,
  }
}
```
which I suspect will cause a problem when the payload gets sent to the timeseries.create api.

Its also possible that I'm misunderstanding how valuetypes and values relate to one another.

Reply:
I see, I misunderstood how that works. I was having trouble locating the ValueType enum, so thanks for sending that. 

So I think what you have here is good. What I would suggest is making a [user-defined type guard](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards) for the functions checking the types. E.g.:

```
function isDistributionValue(value: number | OTDistribution | OTHistogram): value is OTDistribution {
  return value.hasOwnProperty('min');
}
```

Then I think you could safely remove the `as number` type casts in 180 and 182 since it would be narrowed down to number.

That would also help later working with those distribution and histogram types.

Reply:
Great idea! Will do